### PR TITLE
[MMI] adds segment events to Metametrics constants

### PR DESCRIPTION
**Pull request by zone-live**
Adds MMI segment events enums to the Metametrics constants file to be in line with what MetaMask already has.

Segment-schema repo changes:
https://github.com/Consensys/segment-schema/pull/70

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: shared/constants/metametrics.ts [hunk 1]
```diff
@@ -586,7 +586,17 @@ export enum MetaMetricsEventName {
...
   UserClickedCompliance = 'Clicked Compliance',
+  UserClickedRefreshTokenLink = 'User clicked refresh token link',
+  ShowDeeplinkForSignature = 'Show deeplink for signature',
+  CustodianOnboarding = 'Custodian onboarding',
+  InstitutionalFeatureConnection = 'Institutional feature connection',
+  CustodianSelected = 'Custodian selected',
+  ConnectCustodian = 'Connect to custodian',
+  ConnectCustodianCancel = 'Connect to custodian cancel',
+  ConnectCustodianError = 'Connect to custodian error',
+  CustodialAccountsConnected = 'Custodial accounts connected',
```
Comment (brad-decker) [inline shared/constants/metametrics.ts:599]: Event names should all have title casing, and some of these don't make sense as actions. What does 'Custodian Onboarding' mean?. Has @worldlyjohn been asked to validate this schema?

**Review by brad-decker — CHANGES_REQUESTED** [review]

File: shared/constants/metametrics.ts (hunk 1 above)
Comment (worldlyjohn) [reply shared/constants/metametrics.ts:599]: thanks @brad-decker .  i have not, but here are the relevant links.

- these events must be defined in our MM-wide data schema.  it's a new process that we just started implemented. [MetaMask: Working with Analytics/Events](https://docs.google.com/presentation/d/1HuMYZ8nVm-oLiyx8ds-JdcSFQ-_6Jcf2uWcLi43CtZ8/edit#slide=id.g22dbc521061_0_32) (Slide Deck)
- [Event Naming Conventions](https://www.notion.so/Event-Naming-Conventions-95bf27517033451888053f7ab654d327) (Consensys Notion)

Comment (zone-live) [reply shared/constants/metametrics.ts:599]: Hey folks, thanks for the feedback, I'll update the event names to be Title Casing, they are the ones we have in use already. My plan was to add/update them here in the extension and then add them in the [Segment Schema](https://github.com/ConsenSys/segment-schema) repo.
Also I don't have access to the [Event Naming Conventions](https://www.notion.so/Event-Naming-Conventions-95bf27517033451888053f7ab654d327) (Consensys Notion) can you provide it to me?

**Comment by zone-live** [comment]
Hi @worldlyjohn can you give me write access to this repo: [segment-schema](https://github.com/ConsenSys/segment-schema/tree/main)
I'm trying to push a branch there with our events, and I'm seeing the error `Write access to repository not granted.`. 
Thank you! 💪🏼

**Comment by worldlyjohn** [comment]
hey @zone-live please request access via helpdesk@consensys.net

**Comment by zone-live** [comment]
Hi @brad-decker I've updated this PR, with the work done in the [segment-schema](https://github.com/Consensys/segment-schema/pull/70) repo that has been approved by @worldlyjohn.
If you can approve this one now, I'll appreciate! 💪🏼

**Review by brad-decker — APPROVED** [review]

**Review by albertolive — APPROVED** [review]

**Pull request merged**